### PR TITLE
input replacement

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3447794&stafilter=NotComplete&fileview=grid

### CODE REVIEW

File: src/components/forms/purchase-order-form.js [hunk 1]
```diff
@@ -351,18 +356,23 @@ class PurchaseOrderForm extends React.Component {
                                     }}
                                 />
                             </div>
-                            <div className="col-md-2">
+                            <div className="col-md-4">
                                 <label> {T.translate("edit_purchase_order.promo_code")}</label>
-                                <Input
+                                <PromoCodeInput
```
Comment (smarcet) [inline src/components/forms/purchase-order-form.js:361]: use this one https://github.com/OpenStackweb/openstack-uicore-foundation/blob/5cb567f9a4f402bafcfdef7ded2503457a35dc51/src/components/inputs/promocode-input.js#L18

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: src/components/forms/purchase-order-form.js (hunk 1 above)
Comment (romanetar) [reply src/components/forms/purchase-order-form.js]: done

File: src/actions/promocode-actions.js [hunk 2]
```diff
@@ -151,6 +153,23 @@ export const getPromocode = (promocodeId) => async (dispatch, getState) => {
     );
 };
 
+export const queryPromocodes = _.debounce(async (summitId, input, callback) => {
+
+    const accessToken = await getAccessTokenSafely();
+
+    input = escapeFilterValue(input);
+
+    const params = `filter[]=code=@${input}`;
```
[the remark this answers is not in the document]
Comment (romanetar) [reply src/actions/promocode-actions.js]: done

File: src/components/forms/purchase-order-form.js [hunk 3]
```diff
@@ -14,10 +14,10 @@
 import React from 'react'
 import T from 'i18n-react/dist/i18n-react'
 import 'awesome-bootstrap-checkbox/awesome-bootstrap-checkbox.css'
-import { Input, Dropdown, Table, Panel } from 'openstack-uicore-foundation/lib/components'
+import { Input, Dropdown, Table, Panel, PromocodeInput } from 'openstack-uicore-foundation/lib/components'
 import OwnerInput from '../inputs/owner-input'
-import {isEmpty, scrollToError, shallowEqual} from "../../utils/methods";
-
+import {hasErrors, isEmpty, scrollToError, shallowEqual} from "../../utils/methods";
+import {queryPromocodes} from "../../actions/promocode-actions";
```
Comment (smarcet) [inline src/components/forms/purchase-order-form.js:20]: this is not needed

File: src/actions/promocode-actions.js [hunk 4]
```diff
@@ -151,6 +153,19 @@ export const getPromocode = (promocodeId) => async (dispatch, getState) => {
     );
 };
 
+export const queryPromocodes = _.debounce(async (summitId, input, callback) => {
```
Comment (smarcet) [inline src/actions/promocode-actions.js:156]: this is not needed

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**